components/Navigator/actions/submitSearch.js

```diff
@@ -8,64 +8,68 @@
 import { SEARCH_PATH } from '@shopgate/pwa-common-commerce/search/constants';
 import { getSearchPhrase } from '@shopgate/pwa-common/selectors/history';
 import { pushHistory, replaceHistory } from '@shopgate/pwa-common/actions/history/changeHistory';
-import { setSearchPhrase } from '../action-creators';
+import getSearchResults from '@shopgate/pwa-common-commerce/search/actions/getSearchResults';
+import setActiveFilters from '@shopgate/pwa-common-commerce/filter/action-creators/setActiveFilters';
+import setViewTop from 'Components/View/action-creators/setViewTop';
+import setNavigatorSearchPhrase from '../actions/setNavigatorSearchPhrase';
 import toggleNavSearchField from './toggleNavSearchField';
 
  /**
-  * Performs appropriate action(s) when UI search gets triggered.
+  * Performs appropriate action(s) when UI search is submitted.
   * @return {Function} A redux thunk.
   */
 const submitSearch = () => (dispatch, getState) => {
   const state = getState();
   const { searchActive, searchPhrase } = state.navigator;
 
   if (!searchActive) {
-    // Reset search phrase
-    dispatch(setSearchPhrase(''));
-    // Show search input
+    // Reset search phrase.
+    dispatch(setNavigatorSearchPhrase(''));
+    // Show search input.
     dispatch(toggleNavSearchField(true));
     return;
   } else if (!searchPhrase) {
-    // Hide search input
+    // Hide search input.
     dispatch(toggleNavSearchField(false));
     return;
   }
 
-  // Perform search by entering/updating search route
+  // Perform search by entering/updating search route.
   const { history } = state;
   const { sort, ...otherParams } = history.queryParams;
   const prevSearchPhrase = getSearchPhrase(state);
 
+  dispatch(setActiveFilters({}));
```

Comment:
You may consider dispatching this within a subscription.

Reply:
Not in the scope of this ticket.

Reply:
Okay